axelrod/fingerprint.py

```diff
@@ -0,0 +1,206 @@
+            else:
+                edge = (0, index + 2)
+            edges.append(edge)
+        return edges
+
+    def fingerprint(self, turns=50, repetitions=10, granularity=0.01, cores=None):
+        """Build and play a spatial tournament.
+
+        Creates the probes and their edges then builds a spatial tournament
+        where the original strategy only plays probes whose coordinates sum to
+        less than 1 (or equal). Probes whose coordinates sum to more than 1 play
+        the Dual Strategy.
+
+        Parameters
+        ----------
+        turns : integer, optional
+            The number of turns per match
+        repetitions : integer, optional
+            The number of times the round robin should be repeated
+        granularity : float, optional
+            The seperation between each coordinate. Smaller granularity will
+            produce more coordinates that will be closer together.
+        cores : integer, optional
+            The number of processes to be used for parallel processing
+        """
+        self.probe_players = self.create_probes(self.probe, granularity)
+        self.edges = self.create_edges(self.probe_players)
+        original = self.strategy()
+        dual = DualTransformer()(self.strategy)()
+        probes = self.probe_players.values()
+        tourn_players = [original, dual] + list(probes)
+        spatial_tourn = axl.SpatialTournament(tourn_players, turns=turns,
+                                              repetitions=repetitions,
+                                              edges=self.edges)
+        print("Begin Spatial Tournament")
+        self.results = spatial_tourn.play(processes=cores,
+                                          keep_interactions=True)
+        print("Spatial Tournament Finished")
+
+    def _generate_data(self, results, probe_coords):
+        """Generates useful data from a spatial tournament.
+
+        Matches interactions from `results` to their corresponding coordinate in
+        `probe_coords`.
+
+        Parameters
+        ----------
+        results : axelrod.result_set.ResultSetFromFile
+            A results set for a spatial tournament.
+        probe_coords : list of tuples
+            A list of tuples of length 2, where each tuple represents a
+            coordinate, eg. (x, y).
+
+        Returns
+        ----------
+        data_frame : pandas.core.frame.DataFrame
+            A pandas DataFrame object where the row and column headers
+            correspond to coordinates. The cell values are the score of the
+            original/dual strategy playing the probe with parameters that match
+            the coordinate.
+        """
+        edge_scores = {key: cfspt(value[0])[0] for key, value in
+                       results.interactions.items()}
+
+        coord_scores = OrderedDict.fromkeys(probe_coords)
+        for index, coord in enumerate(coord_scores.keys()):
+            if sum(coord) > 1:
```

Comment:
Perhaps a short inline comment describing what's happening with the two cases here (playing the dual or not).